packages/viewer/src/components/scene-tree/readme.md

```diff
@@ -83,26 +95,303 @@ information about this component.
+
+```html
+<html>
+  <body>
+    <vertex-scene-tree viewer-selector="#viewer">
+      <vertex-scene-tree-table-layout>
+        <vertex-scene-tree-table-column>
+          <template>
+            <vertex-scene-tree-table-cell prop:value="{{row.node.name}}" expand-toggle></vertex-scene-tree-table-cell>
+          </template>
+        </vertex-scene-tree-table-column>
+        <vertex-scene-tree-table-column>
+          <template>
+            <vertex-scene-tree-table-cell prop:value="{{row.metadata.PRODUCT_ID}}" visibility-toggle></vertex-scene-tree-table-cell>
+          </template>
+        </vertex-scene-tree-table-column>
+      </vertex-scene-tree-table-layout>
+    </vertex-scene-tree>
+
+    <vertex-viewer id="viewer" src="urn:vertexvis:stream-key:my-key"></vertex-viewer>
+
+    <script type="module">
+      function main() {
+        const tree = document.querySelector('vertex-scene-tree');
+        
+        tree.metadataKeys = [
+          "VERTEX_SCENE_ITEM_NAME",
+          "PRODUCT_ID"
+        ];
+
+        tree.metadataSearchKeys = [
+          "PRODUCT_ID"
+        ];
+      }
+
+      main();
+    </script>
+  </body>
+</html>
+```
+
+In addition to filtering from the default [`<vertex-scene-tree-search>`](../scene-tree-search/readme.md) element, filtering
+can also be done programmatically using the `filterItems` method.
+
+**Example:**
+
+```html
+<html>
+  <body>
+    <vertex-scene-tree viewer-selector="#viewer">
+      <div slot="header">My Custom Header</div>
+
+      <vertex-scene-tree-table-layout>
+        <vertex-scene-tree-table-column>
+          <template>
+            <vertex-scene-tree-table-cell prop:value="{{row.node.name}}" expand-toggle></vertex-scene-tree-table-cell>
+          </template>
+        </vertex-scene-tree-table-column>
+        <vertex-scene-tree-table-column>
+          <template>
+            <vertex-scene-tree-table-cell prop:value="{{row.metadata.PRODUCT_ID}}" visibility-toggle></vertex-scene-tree-table-cell>
+          </template>
+        </vertex-scene-tree-table-column>
+      </vertex-scene-tree-table-layout>
+    </vertex-scene-tree>
+
+    <vertex-viewer id="viewer" src="urn:vertexvis:stream-key:my-key"></vertex-viewer>
+
+    <button id="product-filter-one">Filter to product 1</button>
+    <button id="product-filter-two">Filter to product 2</button>
+
+    <script type="module">
+      function main() {
+        const tree = document.querySelector('vertex-scene-tree');
+        
+        tree.metadataKeys = [
+          "VERTEX_SCENE_ITEM_NAME",
+          "PRODUCT_ID"
+        ];
+
+        const button1 = document.querySelector('#product-filter-one');
+        const button2 = document.querySelector('#product-filter-two');
+
+        button1.addEventListener('click', () => {
+          // Note that the `columns` property must be specified here when programmatically filtering.
+          // See the `FilterTreeOptions` type for additional options when filtering.
```

Comment:
Could there be a troubleshooting section where easily missed things like this could be listed?